e2e/browser/src/components/notifications/index.tsx

```diff
@@ -0,0 +1,149 @@
+import { WebsocketNotification } from "@inrupt/solid-client-notifications";
+import { useEffect, useState } from "react";
+import {
+  createContainerInContainer,
+  getSourceIri,
+  deleteContainer,
+  getPodUrlAll,
+} from "@inrupt/solid-client";
+import { getDefaultSession } from "@inrupt/solid-client-authn-browser";
+
+const MessageList = (props: { messages: Array<any> }) => {
+  const { messages } = props;
+  return (
+    <ul data-testid="eventList">
+      {messages.map((message) => (
+        <li key={message.id}>
+          <pre>{JSON.stringify(message, null, 2)}</pre>
+        </li>
+      ))}
+    </ul>
+  );
+};
+
+export default function Notifications() {
+  const [socket, setSocket] = useState<WebsocketNotification>();
+  const [connectionStatus, setConnectionStatus] = useState<string>(
+    "disconnected"
+  );
+  const [parentContainerUrl, setParentContainerUrl] = useState<string>();
+  const [childContainerUrl, setChildContainerUrl] = useState<string>();
+  const [messageBus, setMessageBus] = useState<any[]>([]);
+
+  useEffect(() => {
+    if (getDefaultSession().info.webId !== undefined) {
+      getPodUrlAll(getDefaultSession().info.webId as string, {
+        fetch: getDefaultSession().fetch,
+      }).then((pods) => {
+        if (pods.length === 0) {
+          throw new Error("No pod root in webid profile");
+        }
+        setParentContainerUrl(pods[0]);
+      });
+    }
+  }, []);
+
+  useEffect(() => {
+    if (parentContainerUrl !== undefined) {
+      setSocket(
+        new WebsocketNotification(parentContainerUrl, {
+          fetch: getDefaultSession().fetch,
+          gateway: "https://notification.inrupt.com",
+        })
+      );
+    }
+  }, [parentContainerUrl]);
+
+  useEffect(() => {
+    if (socket !== undefined) {
+      socket.on("connected", () => setConnectionStatus("connected"));
+      socket.on("closed", () => {
+        setConnectionStatus("closed");
+        setMessageBus([]);
+      });
+      socket.on("error", () => setConnectionStatus("error"));
+      socket.on("message", (message) => {
+        console.debug("New message", { message });
+        messageBus.push(JSON.parse(message));
+        // Reverse the message bus so that the latest message appears first
+        setMessageBus([...messageBus.reverse()]);
```

Comment:
this is odd, you'd be better off doing:
```suggestion
        const parsedMessage = JSON.parse(message);
        setMessageBus([parsedMessage, ...messageBus]);
```

Reply:
That's what I initially did, but for some reason I don't understand, written like this the message array only ever contains one element, the latest message.

Reply:
hmm, maybe:
```
setMessageBus((previousMessageBus) => [parsedMessage, ...previousMessageBus]);
```

per https://reactjs.org/docs/hooks-reference.html#functional-updates

Reply:
Aha, that does the trick, thanks ! 246e4557b44aff30affeca6d04e052b76f564a9b